### PR TITLE
Query: Adds simple redundant SelectExpression join elimination.

### PR DESCRIPTION
**Pull request by anpete**
Fix: #8372

### CODE REVIEW

File: src/EFCore.Relational/Query/Expressions/SelectExpression.cs [hunk 1]
```diff
@@ -570,6 +571,20 @@ var projectionIndex
         }
 
         /// <summary>
+        ///     TODO
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/Expressions/SelectExpression.cs]: summary

File: src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs [hunk 2]
```diff
@@ -268,14 +268,169 @@ public override void VisitQueryModel(QueryModel queryModel)
...
+
+                    if (tableExpression is LeftOuterJoinExpression joinExpressionBase)
+                    {
+                        _tables.Clear();
+                        _columns.Clear();
+
+                        Visit(joinExpressionBase.Predicate);
+
+                        if (_columns.Count == 1
+                            && _tables.Count == 2
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: Does this work with Composite key case?

File: src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs [hunk 3]
```diff
@@ -268,14 +268,169 @@ public override void VisitQueryModel(QueryModel queryModel)
...
+                            && _tables.Distinct(_tableColumnEqualityComparer).Count() == 1)
+                        {
+                            var newTableExpression
+                                = _tables.Single(t => !ReferenceEquals(t, joinExpressionBase.TableExpression));
+
+                            selectExpression.RemoveTable(joinExpressionBase);
+
+                            if (ReferenceEquals(selectExpression.ProjectStarTable, joinExpressionBase)
+                                || ReferenceEquals(selectExpression.ProjectStarTable, joinExpressionBase.TableExpression))
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: It should work with just joinExpressionBase (no need to unwrap) But if something is failing due to that, just file an issue for me to investigate.

File: src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs [hunk 4]
```diff
@@ -268,14 +268,169 @@ public override void VisitQueryModel(QueryModel queryModel)
...
+                        && y is TableExpression tableExpressionY)
+                    {
+                        return tableExpressionX.Table == tableExpressionY.Table
+                               && tableExpressionX.Schema == tableExpressionY.Schema;
+                    }
+
+                    return false;
+                }
+
+                public int GetHashCode(TableExpressionBase obj) => 0;
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: Intentional?

Comment (anpete) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: Yeah, forcing Equals here.

File: src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs (hunk 3 above)
Comment (anpete) [inline src/EFCore.Relational/Query/RelationalQueryModelVisitor.cs]: Filed #8412

**Pull request merged**